azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecutorServer.java

```diff
@@ -260,6 +263,28 @@ private void startReportingExecMetrics() {
     this.metricsManager.startReporting("AZ-EXEC", this.props);
   }
 
+  private void initActive() throws ExecutorManagerException {
+    final Executor executor;
+    try {
+      final String host = requireNonNull(getHost());
+      // TODO "pick any free port" can't be supported like this because it's only available after the
+      // jetty server has been started. this works if executor.port is set
+      final int port = props.getInt(ConfigurationKeys.EXECUTOR_PORT, -1);
+      checkState(port != -1);
```

Comment:
Fixed port is not possible as there can be multiple executors on same host. A new deployment lets all the new executions to take place on the latest executor, however, all the existing ones continue to use the executor on which they started. These executor instances shutdown only when they are drained out.
Can you elaborate on the if condition?
What should be condition?

Reply:
> What should be condition?

If port <= 0, just log a warning that FlowRunnerManager will be initialized in inactive state. The log message can also mention that FlowRunnerManager can be activated manually via the API (which is needed any way when creating an entirely new executor), or then a fixed executor.port should be set so that the state can be checked from DB before initializing the state.